src/couch_expiring_cache/src/couch_expiring_cache.erl

```diff
@@ -0,0 +1,33 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_expiring_cache).
+
+-export([
+    lookup/2,
+    insert/5
+]).
+
+
+-include_lib("couch_expiring_cache/include/couch_expiring_cache.hrl").
+
+
+-spec insert(Name :: binary(), Key :: binary(), Value :: binary(),
+    StaleTS :: millisecond(), ExpiresTS :: millisecond()) -> ok.
+insert(Name, Key, Value, StaleTS, ExpiresTS) ->
+    couch_expiring_cache_fdb:insert(Name, Key, Value, StaleTS, ExpiresTS).
+
+
+-spec lookup(Name :: binary(), Key :: binary()) ->
+    not_found | {fresh, Val :: binary()} | {stale, Val :: binary()} | expired.
+lookup(Name, Key) ->
```

Comment:
I like the idea that @nickva mentions above about adding an optional transaction. I think it should also apply to the lookup.

Reply:
https://github.com/apache/couchdb/pull/2331/commits/b9392164bf6f86dc3f2906959c9226ab55e6e347 enables optional transactions for both insert and lookup operations.